userspace/libscap/CMakeLists.txt

```diff
@@ -152,57 +194,14 @@ if(CREATE_TEST_TARGETS AND NOT WIN32)
 		add_subdirectory(test)
 endif()
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-	add_subdirectory(linux)
-	include_directories(linux)
-elseif(WIN32)
-	add_subdirectory(win32)
-	include_directories(win32)
-elseif(APPLE)
-	add_subdirectory(macos)
-	include_directories(macos)
-endif()
-
 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
```

Comment:
IMHO the best approach here is to add a `driver_event_schema` library with syscall_table and fillers_table, on linux link both the `scap_event_schema` and `driver_event_schema`, and on non-linux just linking `scap_event_schema`; wdyt?
It would save us all other changes in the cmake files and it's better!

With you changes, the `scap_event_schema` library becomes useless (i am thinking about eg: modern bpf tests that link it).

Reply:
@leogr @Andreagit97 wdyt?

Reply:
Makes sense to me. Done in the current version of the PR.

Switching to object libraries is still needed, at least for scap_savefile_engine and scap_source_plugin_engine. Without that I get:
```
Undefined symbols for architecture x86_64:
  "_scap_savefile_engine", referenced from:
      _scap_open_offline_int in libscap.a(scap.c.o)
  "_scap_source_plugin_engine", referenced from:
      _scap_open_plugin_int in libscap.a(scap.c.o)
```

Reply:
Mmh that is weird :/ 
I mean, linking libraries should work on windows too, without using objects instead. I don't understand why it should be failing.

Reply:
Once we found the best solution, i think we should be able to run `make sinsp-example` on windows CI, to be able to test the build and linking of objects (and the same for macos).
This way, no future breakage of this will happen :)

Reply:
Yeah, we absolutely need to run `make sinsp-example` in order to avoid future regressions! I agree in splitting this into 2 libraries  `scap_event_schema` and `driver_event_schema`